src/ng/directive/ngOptions.js

```diff
@@ -467,9 +467,15 @@ var ngOptionsDirective = ['$compile', '$document', '$parse', function($compile,
       if (!multiple) {
 
         selectCtrl.writeValue = function writeNgOptionsValue(value) {
+          var selectedOption = options.selectValueMap[selectElement.val()];
           var option = options.getOptionFromViewValue(value);
 
+          // Make sure to remove the selected attribute from the previously selected option
+          // Otherwise, screen readers might get confused
+          if (selectedOption) selectedOption.element.removeAttribute('selected');
+
           if (option) {
+            if (providedEmptyOption) emptyOption.removeAttr('selected');
```

Comment:
I guess not, because I think when the empty option is removed and then re-added, we need it to be selected. But moving it into the fn it saves precious CPU cycle!